fml/platform/darwin/scoped_typeref.h

```diff
@@ -119,10 +128,11 @@ class ScopedTypeRef {
     object_ = temp;
   }
 
+ protected:
   // ScopedTypeRef<>::release() is like std::unique_ptr<>::release.  It is NOT
   // a wrapper for Release().  To force a ScopedTypeRef<> object to call
   // Release(), use ScopedTypeRef<>::reset().
-  T release() __attribute((ns_returns_not_retained)) WARN_UNUSED_RESULT {
```

Comment:
If want to use `[[nodiscard]]` instead of the macro, presumably we should remove the part of this PR that adds the macro to our codebase and replace any other usage as well?